app/lib/discourse_automation/triggers/recurring.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+DiscourseAutomation::Triggerable::RECURRING = 'recurring'
+
+RECURRENCE_CHOICES = [
+  { id: 'every_minute', name: 'discourse_automation.triggerables.recurring.recurrences.every_minute' },
+  { id: 'every_hour', name: 'discourse_automation.triggerables.recurring.recurrences.every_hour' },
+  { id: 'every_day', name: 'discourse_automation.triggerables.recurring.recurrences.every_day' },
+  { id: 'every_weekday', name: 'discourse_automation.triggerables.recurring.recurrences.every_weekday' },
+  { id: 'every_week', name: 'discourse_automation.triggerables.recurring.recurrences.every_week' },
+  { id: 'every_other_week', name: 'discourse_automation.triggerables.recurring.recurrences.every_other_week' },
+  { id: 'every_month', name: 'discourse_automation.triggerables.recurring.recurrences.every_month' },
+]
+
+def setup_pending_automation(automation, fields)
+  automation.pending_automations.destroy_all
+
+  expected_recurrence = fields.dig('recurrence', 'value')
+  return if !expected_recurrence
+
+  case expected_recurrence
+  when 'every_day'
+    next_trigger_date = RRule::Rule
+      .new('FREQ=DAILY', dtstart: Time.now)
+      .between(Time.now, Time.now + 2.days)
+      .first
+  when 'every_month'
+    next_trigger_date = RRule::Rule
+      .new('FREQ=MONTHLY', dtstart: Time.now)
+      .between(Time.now, Time.now + 2.months)
+      .first
+  when 'every_weekday'
+    next_trigger_date = RRule::Rule
+      .new('FREQ=DAILY;BYDAY=MO,TU,WE,TH,FR', dtstart: Time.now)
+      .between(Time.now.end_of_day, Time.now + 3.days)
+      .first
+  when 'every_week'
+    next_trigger_date = RRule::Rule
+      .new('FREQ=WEEKLY;INTERVAL=1', dtstart: Time.now)
```

Comment:
Just curious, do you need the `:INTERVAL=1` when you don't need it for `DAILY` and `MONTHLY`?

Reply:
hmm no indeed that's a mistake, should do anything bad but yes probably not needed here :+1: will check